src/ring_theory/integrally_closed.lean

```diff
@@ -36,7 +36,7 @@ class is_integrally_closed (R : Type*) [comm_ring R] [is_domain R] : Prop :=
 section iff
 
 variables {R : Type*} [comm_ring R] [is_domain R]
-variables (K : Type*) [field K] [algebra R K] [is_fraction_ring R K]
```

Comment:
Why did you modified `field K`? My suggestion is to keep it, but introducing another field, that is `L` in my example, and I think `is_field L` can be relaxed.

Reply:
Why should we require `field K` in general when `comm_ring K` is sufficient? This doesn't seem to break anything elsewhere.

Reply:
See the explanation [here](https://leanprover-community.github.io/mathlib_docs/ring_theory/localization/basic.html). Essentially there will be no instance of `field` on `K`, even if of course it is a field. We cannot have that def an instance because we would have a diamond, so it's better to have `field K` from the beginning.

Reply:
I see, I didn't know about this.